### ISSUE TITLE
CoreML export failure: 'ConvModule' object has no attribute 'norm'

### ISSUE THREAD

**Issue by ghoshaw** [RangiLyu/nanodet#206]
Hi, I tried to turn the nanodet-m.pth to coreml for IOS. I used coremltools as the guide, and got error "CoreML export failure: 'ConvModule' object has no attribute 'norm'". I read the source code of nanodet found that the norm in head is BN which should be supported by coreml. So I do not know why is the error happening. Is anyone has tried coreml?
Thanks!

**Comment by batrlatom** [comment]
Would you please try this? 

```
model = predictor.model.cpu()
print(model)
img = torch.ones((1, 3, 416, 416)).cpu()
ts = torch.jit.trace(model, img)

torch.jit.save(ts, 'nanodet.jit.ts')
ts = torch.jit.load('nanodet.jit.ts')


try:
      import coremltools as ct

      print('\nStarting CoreML export with coremltools %s...' % ct.__version__)
      # convert model from torchscript and apply pixel scaling as per detect.py
      model = ct.convert(ts, inputs=[ct.ImageType(name='image', shape=img.shape, scale=1 / 255.0, bias=[0, 0, 0])])
      #f = opt.weights.replace('.pt', '.mlmodel')  # filename
      model.save('/content/nanodet.mlmodel')
      print('CoreML export success, saved as %s' % f)
except Exception as e:
      print('CoreML export failure: %s' % e)

```

**Comment by zxzheng826** [comment]
I found that this problem is inherited from the project OpenMMDetection(mmcv).  Change the source code in Nanodet project can help me export torchscript model. Maybe it can help you too. https://github.com/open-mmlab/mmcv/pull/889/files

**Comment by ghoshaw** [comment]
Thanks, this problem is solved. And I found that the turned model in coreml is coreml v5 which can only run above IOS14, maybe who knows how to turn a coreml v4 version model? 
![image](https://user-images.githubusercontent.com/23494873/114959133-053bfd80-9e97-11eb-98c8-754200982806.png)

**Comment by minushuang** [comment]
hi @ghoshaw, did you convert mlmodel sucessed? I convert the nanodet model from torch to coreml, and  test the coreml model with some images, the results are not correct, seems loss precise.

here is my convert code.
```
def main(config, model_path, output_path, input_shape=(320, 320)):
    logger = Logger(-1, config.save_dir, False)
    model = build_model(config.model)
    checkpoint = torch.load(model_path, map_location=lambda storage, loc: storage)
    load_model_weight(model, checkpoint, logger)

    dummy_input = torch.autograd.Variable(
        torch.randn(1, 3, input_shape[0], input_shape[1])
    )

    traced_model = torch.jit.trace(model, dummy_input)

    logging.info("convert coreml start.")
    core_model = ct.convert(
        traced_model,
        inputs=[ct.ImageType(shape=dummy_input.shape, name='input', scale=0.017429, bias=(-103.53 * 0.017429, -116.28 * 0.017507, -123.675 * 0.017125))],
        outputs=[ct.TensorType(name="output")],
        debug=True
    )
    core_model.save(output_path)
    logging.info("finish convert coreml.")
```
coreml infer code
```
example_image = Image.open(img_path).resize((320, 320)).convert('RGB')
model = ct.models.MLModel(mlmodel_path)
preds = model.predict({'input': example_image})
#post-process nms for preds
```
torch results
![image](https://user-images.githubusercontent.com/5811852/198556739-bf7a189d-7605-486c-9a4b-7c144c27d58e.png)
coreml model results
![image](https://user-images.githubusercontent.com/5811852/198556800-26c2650e-3829-4468-bab5-69908ba1f8e2.png)

any help will be appreciate.

**Comment by ghoshaw** [comment]
It seems ok to me, maybe you should check if the preprocess like mean and std or RGB is set correctly?

**Comment by minushuang** [comment]
> It seems ok to me, maybe you should check if the preprocess like mean and std or RGB is set correctly.

thank you for your kind response, I forgot to set the model to eval mode before converting, it's ok now after `model.eval()`
